src/main/java/org/apache/mesos/scheduler/plan/DefaultPlanCoordinator.java

```diff
@@ -0,0 +1,82 @@
+package org.apache.mesos.scheduler.plan;
+
+import com.google.common.annotations.VisibleForTesting;
+import org.apache.commons.collections.CollectionUtils;
+import org.apache.mesos.Protos;
+import org.apache.mesos.SchedulerDriver;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.*;
+import java.util.stream.Collectors;
+
+/**
+ * Default implementation of PlanCoordinator.
+ */
+public class DefaultPlanCoordinator implements PlanCoordinator {
+    private static final Logger LOGGER = LoggerFactory.getLogger(DefaultPlanCoordinator.class);
+
+    private List<PlanManager> planManagers = new LinkedList<>();
+    private PlanScheduler planScheduler;
+
+    public DefaultPlanCoordinator(List<PlanManager> planManagers, PlanScheduler planScheduler) {
+        if (CollectionUtils.isEmpty(planManagers)) {
+            throw new IllegalArgumentException("Atleast one plan manager is required");
+        }
+        this.planManagers.addAll(planManagers);
+        this.planScheduler = planScheduler;
+    }
+
+    @Override
+    public Collection<Protos.OfferID> processOffers(final SchedulerDriver driver,
+                                                    final List<Protos.Offer> offersToProcess) {
+        final Set<Protos.OfferID> dirtiedOffers = new HashSet<>();
+        final List<Block> dirtiedAssets = new ArrayList<>();
+        final List<Protos.Offer> offers = new ArrayList<>(offersToProcess);
+        for (final PlanManager planManager : planManagers) {
+            try {
+                LOGGER.info("Current plan {} interrupted.", planManager.isInterrupted() ? "is" : "is not");
```

Comment:
Can plans (or plan managers) have a name? Would be useful for debugging. Could then log the name of the plan here and below.